### DISCUSSION TITLE
[Q] Is rr_http_workers_memory_bytes the only metric available?

### DISCUSSION THREAD

**Discussion by victor-sudakov** [spiral/roadrunner-plugins#102] [Q&A] [answered]
Hello!

I've enabled Prometheus metrics in rr with
```
metrics:
  address: 0.0.0.0:2112
```
and added `"spiral/roadrunner-metrics": "^2.0.1"` to composer.json. The only metric I see in Prometheus is "rr_http_workers_memory_bytes". Is it really the only builtin metric in rr? I'd like to see more info, like number of client connections, worker execution times etc. Is that possible?

This Issue should be better classified as a help request rather than a bug.

The docker images are spiralscout/roadrunner:2.4.2 and composer:2.1.3

**Comment by rustatian** [comment, marked as answer]
Hey @victor-sudakov . To see the other metrics, you need to perform at least one request (HTTP).

**Reply by victor-sudakov** [reply]
Interesting. Thank you!

**Comment by rustatian** [comment]
```
rr_http_request_duration_seconds_bucket
rr_http_request_duration_seconds_sum{status="200"}
rr_http_request_duration_seconds_count{status="200"} 2
rr_http_request_total{status="200"} 2 (+other statuses)
rr_http_workers_memory_bytes"
```

**Reply by victor-sudakov** [reply]
If I have multiple rr pods in a deployment, and Prometheus is scraping a service loadbalancing between those multiple rr pods, how would you advise to aggregate those statistics in PromQL to see the whole picture of the load? Just sum everything up from each pod or what?

**Reply by rustatian** [reply]
Yeah, if you want to see the full picture, just sum these metrics.

**Reply by victor-sudakov** [reply]
Do you have an example of Roadrunner Prometheus monitoring job at hand, or maybe a link thereto? You probably need to discover all Roadrunner pods behind a service/deployment as separate components, otherwise only a partial picture will be available.

**Reply by rustatian** [reply]
No, I don't have that link. No, you don't need to discover, because Prometheus works by a pull model, all nodes should be in the configuration (or address ranges). If the node has the Prometheus port allocated, Prometheus will get the metrics, otherwise, it'll skip it.

**Reply by victor-sudakov** [reply]
> all nodes should be in the configuration (or address ranges)

AFAIK Prometheus ports are allocated on pods, not on nodes, and pods are dynamic (with dynamic names) within a deployment, so you need some form of discovery. If you don't do Roadrunner monitoring with Prometheus  and don't have a Prometheus config snippet at hand, I'll look elsewhere. Thanks for your help!
